dams/src/crypto.rs

```diff
@@ -85,7 +83,11 @@ impl AssociatedData {
     }
 
     fn with_str(self, ad: &str) -> Self {
-        AssociatedData(format!("{}\n{}", self.0, ad))
+        AssociatedData(self.0.into_iter().chain(ad.bytes().into_iter()).collect())
```

Comment:
```suggestion
        AssociatedData(self.0.into_iter().chain(ad.bytes()).collect())
```
I think the `Bytes` type that `bytes()` returns is already an iterator.

Reply:
Good call. I actually switched this to call into the `with_bytes` method for consistency.